src/libs/kata-sys-util/src/protection.rs

```diff
@@ -177,6 +179,19 @@ pub fn arch_guest_protection(
         };
 
         return Ok(GuestProtection::Tdx(details));
+    } else {
+        // Check if /sys/module/kvm_intel/parameters/tdx is set to 'Y'
+        if Path::new(TDX_KVM_PARAMETER_PATH).exists() {
+            if let Ok(content) = fs::read(TDX_KVM_PARAMETER_PATH) {
+                if !content.is_empty() && content[0] == b'Y' {
+                    // TODO: hardcode with tdx details(1.5) is acceptable ?
```

Comment:
Is there a need for `TDXDetails`?  The implementation isn't supported by any existing TDX host configuration so perhaps drop it altogether.

Reply:
Yeah, I did check the commit for it. It's based on some old TDX host enablement that is not valid with the going-to-be upstream code.

Reply:
@mythi That's to say, we should move it (`TDXDetails`) into the deprecated list ? 
Honestly, I have only one old Tdx platform, and I have ever checked the `TDX_KVM_PARAMETER_PATH` existation, but it's disappear when I reinstall the host with a newer Ubuntu OS(24.04 with kernel 6.8.xxx). If the upstream code without `TDXDetails`, I prefer removing such configuration.

Reply:
The question I had is still open: is there a need for the details? If it's just a debug print, perhaps no deprecation is needed

Reply:
If so, I think we'd better keep it as it was, I just make it detailed with 1.0